src/commands/install.yml

```diff
@@ -18,39 +18,62 @@ steps:
-        if [ "<< parameters.version >>" = "latest" ] && command -v secrethub >/dev/null 2>&1; then
-            exit 0
-        fi
+        set -e
+
+        # Colors
+        NO_COLOR="\033[0m"
+        OK_COLOR="\033[32;01m"
+        ERROR_COLOR="\033[31;01m"
+        WARN_COLOR="\033[33;01m"
 
-        # Get the latest version.
-        VERSION="<< parameters.version >>"
-        if [ "$VERSION" = "latest" ]; then
-            VERSION=$(curl -sSfL "https://api.github.com/repos/secrethub/secrethub-cli/releases/latest" | grep tag_name | awk -F\" '{ print substr($4,2) }')
+        # Detect Architecture
+        ARCH=amd64
+        if [ $(getconf LONG_BIT) = 32 ]; then
+            ARCH=386
         fi
 
-        # Get the current platform.
-        if uname -a | grep Darwin > /dev/null 2>&1; then
-            PLATFORM=darwin
+        # Detect OS
+        UNAME=$(uname)
+        if [ "$UNAME" = "Darwin" ]; then
+          OS=darwin
+        elif [ "$UNAME" = "Linux" ]; then
+            OS=linux
         else
-            PLATFORM=linux
+            echo -e "${ERROR_COLOR}Cannot determine OS type. Exiting...${NO_COLOR}"
+            exit;
+        fi
+
+        # Make sure we have root priviliges.
+        SUDO=""
+        if [ $(id -u) -ne 0 ]; then
+            if ! [ $(command -v sudo) ]; then
+                echo -e "${ERROR_COLOR}Installer requires root privileges. Please run this script as root.${NO_COLOR}"
+                exit;
+            fi
+
+            SUDO="sudo"
         fi
 
-        # If the CLI is not installed or the current version of the CLI does not match the specified one, download the provided version.
-        if ! command -v secrethub >/dev/null 2>&1 || [[ $(secrethub --version 2>&1 | cut -d "," -f 1) != *"$VERSION" ]]; then
-          mkdir secrethub
-          curl -sSfL https://github.com/secrethub/secrethub-cli/releases/download/v$VERSION/secrethub-v$VERSION-$PLATFORM-amd64.tar.gz | tar zxf - -C ./secrethub
-
-          # check if we can install the CLI without sudo
-          if ! mv ./secrethub/bin/secrethub << parameters.path >> 2>/dev/null ; then
-            sudo mv ./secrethub/bin/secrethub << parameters.path >>
-          fi
-          rm -rf ./secrethub
-
-          # Add executable to path if it is not there already.
-          if ! echo ":$PATH:" | grep -q  ":<<parameters.path>>:"; then
-            echo 'export PATH="$PATH:<<parameters.path>>"' >> $BASH_ENV;
-          fi
+        echo -e "${OK_COLOR}==> Creating directories${NO_COLOR}"
+        $SUDO mkdir -p /usr/local/secrethub/bin
+
+        # Retrieve latest version
+        if [ -n "${SECRETHUB_CLI_VERSION}" ]; then
+          VERSION=v${SECRETHUB_CLI_VERSION}
+        else
+          echo -e "${OK_COLOR}==> Retrieving latest version${NO_COLOR}"
+          VERSION=$(curl --silent "https://api.github.com/repos/secrethub/secrethub-cli/releases/latest" | grep tag_name | awk -F\" '{ print $4 }')
         fi
+
```

Comment:
This previous iteration of the install script had a check to not re-install if the requested version is already installed. Do we want to keep that?

Reply:
For now I'd just prefer a more consistent and robust install script. Let's add that later on.